core/modules/user/user.module

```diff
@@ -1094,34 +1094,32 @@ function user_register_access() {
 /**
  * User view access callback.
  *
- * @param $account
+ * @param User|AnonymousUser|object $account
  *   Can either be a full user object or a $uid.
+ *
+ * @return bool
+ *   Whether access is allowed or not.
  */
 function user_view_access($account) {
-  $uid = is_object($account) ? $account->uid : (int) $account;
-
-  // Never allow access to view the anonymous user account.
-  if ($uid) {
-    // Admins can view all, users can view own profiles at all times.
-    if ($GLOBALS['user']->uid == $uid || user_access('administer users')) {
-      return TRUE;
-    }
-    elseif (user_access('access user profiles')) {
-      // At this point, load the complete account object.
-      if (!is_object($account)) {
-        $account = user_load($uid);
-      }
-      return (is_object($account) && $account->status);
-    }
+  if (!($account instanceof User) && is_int($account)) {
+    $account = user_load($account);
   }
-  return FALSE;
+  return $account && $account->access('view');
 }
 
 /**
  * Access callback for user account editing.
  */
 function user_edit_access($account) {
-  return (($GLOBALS['user']->uid == $account->uid) || user_access('administer users')) && $account->uid > 0;
+  if (!($account instanceof User) && is_int($account)) {
+    $account = user_load($account);
+  }
+  // Account for anonymous users, which are of class AnonymousUser.
+  if ($account instanceof User) {
+    /** @var User $account */
```

Comment:
Typehinting for IDEs is good. Though `/**` should just be `/*`. Is the double star syntax created automatically by your IDE?

Reply:
http://docs.phpdoc.org/references/phpdoc/tags/var.html
A single * is not valid.

Reply:
I've updated our docs :) https://api.backdropcms.org/doc-standards#var

Reply:
Looks like you're right @moonray. PHPStorm originally only supported `/**` but supported `/*` later. NetBeans has it's own syntax that is similar but only uses `/*` and the syntax is reversed. https://stackoverflow.com/questions/1816641/jetbrains-webide-php-variable-type-hinting

In any case, `/**` seems to be preferred as proper docblock, so :+1: